data-prepper-core/src/test/java/com/amazon/dataprepper/pipeline/PipelineTests.java

```diff
@@ -45,6 +46,27 @@ public void testPipelineState() {
         assertThat("Pipeline isStopRequested is expected to be false", testPipeline.isStopRequested(), is(false));
         testPipeline.shutdown();
         assertThat("Pipeline isStopRequested is expected to be true", testPipeline.isStopRequested(), is(true));
+        assertThat("Sink shutdown should be called", testSink.isShutdown, is(true));
+    }
+
+    @Test
+    public void testPipelineStateWithProcessor() {
+        final Source<Record<String>> testSource = new TestSource();
+        final TestSink testSink = new TestSink();
+        final TestProcessor testProcessor = new TestProcessor();
+        final Pipeline testPipeline = new Pipeline(TEST_PIPELINE_NAME, testSource, new BlockingBuffer(TEST_PIPELINE_NAME),
+                Collections.singletonList(testProcessor),
+                Collections.singletonList(testSink),
+                TEST_PROCESSOR_THREADS, TEST_READ_BATCH_TIMEOUT);
+        assertThat("Pipeline isStopRequested is expected to be false", testPipeline.isStopRequested(), is(false));
+        assertThat("Pipeline is expected to have a default buffer", testPipeline.getBuffer(), notNullValue());
+        assertTrue("Pipeline processors should be empty", testPipeline.getProcessors().size() == 1);
+        testPipeline.execute();
+        assertThat("Pipeline isStopRequested is expected to be false", testPipeline.isStopRequested(), is(false));
+        testPipeline.shutdown();
+        assertThat("Pipeline isStopRequested is expected to be true", testPipeline.isStopRequested(), is(true));
+        assertThat("Sink shutdown should be called", testSink.isShutdown, is(true));
+        assertThat("Processor shutdown should be called", testProcessor.isShutdown, is(true));
```

Comment:
Another option to verify this behavior would be to mock out the processor and sink, and call `verify(), once` on the `shutdown()` methods. Could potentially save on having to create test classes to implement the interfaces.

Reply:
Yeah, agreed. I prefer to use the mocks, but in the interest of time, I just set it up to keep in line with how the rest of the tests worked in this test class